app/src/main/java/org/groundplatform/android/data/local/room/stores/RoomLocationOfInterestStore.kt

```diff
@@ -48,9 +53,26 @@ class RoomLocationOfInterestStore @Inject internal constructor() : LocalLocation
     locationOfInterestDao.countByDeletionState(surveyId, EntityDeletionState.DEFAULT)
 
   override fun getValidLois(survey: Survey): Flow<Set<LocationOfInterest>> =
-    locationOfInterestDao.getByDeletionState(survey.id, EntityDeletionState.DEFAULT).map {
-      toLocationsOfInterest(survey, it)
-    }
+    locationOfInterestDao
+      .getByDeletionState(survey.id, EntityDeletionState.DEFAULT)
+      .map { toLocationsOfInterest(survey, it) }
+      .map { lois ->
+        // Filter out LOIs with invalid/empty geometries to prevent crashes
+        lois
+          .filter { loi ->
```

Comment:
Let's move this filter to LOIRepository instead. The purpose of Room*Store is to fetch data from local storage only.